### DISCUSSION TITLE
Preview and render without executing code

### DISCUSSION THREAD

**Discussion by cjackst11** [quarto-dev/quarto-cli#6037] [Q&A] [answered]
### Description

Hello 

I am using Quarto for developing  coding assignments.  It is really fantastic, but there one thing I can't do.

I need to make pdf/html with code only, accomplished by \`\`\`r, but develop the quarto file with \`\`\`{r} so code executes to make sure my code is correct. At the moment, I search and replace all \`\`\`{r} for \`\`\`r, but is there some command line switch that makes that automatic?

Thanks for this wonderful program. 

best Jack

**Comment by cderv** [comment, marked as answer]
You can control execution using some YAML option. For example, you can deactivate all evaluation using 
```
execute:
  eval: false
```
See https://quarto.org/docs/computations/execution-options.html

Another option would be to use a Project, and the `freeze` option: https://quarto.org/docs/projects/code-execution.html#freeze
This will render the document once with execution, and then prevent execution unless you explicitly render. 

It depends what you need exactly - possibly `eval: false` is what you need here !

**Comment by cjackst11** [comment]
Thanks @cderv !

This actually is not what I was looking for, I want code to execute in VScode, so I can make sure it it correct. I just don't want code to execute when I render/preview the files, and only show the code. Reason is that the html/pdf files ultimately go to students and the TAs, and they are to execute the code themselves.  So am really after a command line switch, not a yaml switch.  This is easy to do with a small sed script on the command line for render, (not for preview) but that is very clumcy. 

At the moment I am using a ```on file change``` command to change all the {r} to r and make a new file, but that is super clumcy. 

best jd

**Reply by mcanouil** [reply]
How `eval: false` does not fit what you want?
Code cells can still be executed manually in RStudio or VSCode, while when you render/preview the code is not evaluated.

It seems you want to render for yourself with `eval: true` and for the others with `eval: false`.
If that's the case, you can use project profiles, see <https://quarto.org/docs/projects/profiles.html>.

**Comment by cjackst11** [comment]
@mcanouil Yes, I should have had my coffee before responding. That makes perfect sense. I am still coming to grips with Quarto, and this eventuality escaped me.

And Quarto is truly fantastic. 

best, j
